R/create_covariate_summary_plot.R

```diff
@@ -0,0 +1,158 @@
+#' Create the covariate summary plot 
+#' https://rdrr.io/github/audrey-b/BUGSnet/man/data.plot.html
+#'
+#' @param all_data Study data including covariate columns, in wide or long format
+#' @param metaoutcome Reactive containing meta analysis outcome: "Continuous" or "Binary"
+#' @param covariate_or_baseline Output from widget to toggle between covariate and baseline risk for the plot type.
+#' @param treatment_df Reactive containing data frame containing treatment IDs (Number) and names (Label)
+#' @return BUGSnet::data.plot plot
+
+CreateCovariateSummaryPlot <- function(all_data, metaoutcome, covariate_or_baseline, treatment_df) {
```

Comment:
This function is 128 lines long, can it be broken up into smaller functions please? Maybe along lines of covariate/baseline and/or continuous/binary?

Reply:
Taken out the plot settings and data mutation into separate functions, which makes the remainder more readable

Reply:
Can you update the function names please? Because R has a global namespace, simple general function names are too likely to clash with other functions, either our own, or those in a package.

Something like `CreateCovariateSummaryPlotSettings` and `MutateCovariateSummaryData` would be safer.

I wish we could have simpler names, but this is safer.